internal/fs/fs.go

```diff
@@ -1670,33 +1673,37 @@ func (fs *fileSystem) createLocalFile(
 		}
 	}()
 
-	fs.mu.Lock()
-
 	fullName := inode.NewFileName(parent.Name(), name)
-	child, ok := fs.localFileInodes[fullName]
+	chld, ok := fs.localFileInodes[fullName]
+	child = chld
 
 	if ok && !child.(*inode.FileInode).IsUnlinked() {
 		return
 	}
 
 	// Create a new inode when a file is created first time, or when a local file is unlinked and then recreated with the same name.
-	var result *inode.Core
-	result, err = parent.CreateLocalChildFile(name)
-	if err != nil {
-		return
-	}
-
-	child = fs.mintInode(*result)
+	core := parent.LocalChildFileCore(name)
+	child = fs.mintInode(core)
 	fs.localFileInodes[child.Name()] = child
 
 	// Empty file is created to be able to set attributes on the file.
 	fileInode := child.(*inode.FileInode)
-	err = fileInode.CreateEmptyTempFile()
-	if err != nil {
-		return
+	if err := fileInode.CreateEmptyTempFile(); err != nil {
+		return nil, err
 	}
-
-	return
+	fs.mu.Unlock()
+	parent.Lock()
+	defer func() {
+		if err != nil {
```

Comment:
I didnt follow this. If err is not nil, you are returning from line 1692. What error will come here?